persistence-modules/spring-data-redis/src/test/java/com/baeldung/spring/data/redis_ttl/LibraryIntegrationTest.java

```diff
@@ -0,0 +1,93 @@
+package com.baeldung.spring.data.redis_ttl;
+
+import java.io.IOException;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.baeldung.spring.data.redis_ttl.entity.Book;
+import com.baeldung.spring.data.redis_ttl.entity.Librarian;
+import com.baeldung.spring.data.redis_ttl.repository.BookRepository;
+import com.baeldung.spring.data.redis_ttl.repository.LibrarianRepository;
+
+import redis.embedded.RedisServerBuilder;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = LibraryApplication.class)
+@DirtiesContext(classMode = ClassMode.BEFORE_CLASS)
+public class LibraryIntegrationTest {
+    
+    private static redis.embedded.RedisServer redisServer;
+    
+    @BeforeClass
+    public static void startRedisServer() throws IOException {
+        redisServer = new RedisServerBuilder().port(6379).setting("maxmemory 256M").build();
```

Comment:
This runs the risk of port clashing with anything else running on Jenkins. Perhaps this test shouldn't be run at all in the Baeldung CI Build?

Reply:
Not really sure about this. I actually looked at other modules under spring-data-redis and used the same approach.
Any suggestion that you have to on how to test this ?